### PR TITLE
Enhanced the explanation of subdomain multi-tenancy

### PR DESCRIPTION
**Pull request by Adam-Langley**
The previous document appears to be written for scenarios where the Angular and API projects were merged. 
These changes should more clearly cover both scenarios, and also explains how tenant names are resolved at runtime, should the developer experience any issues.

### CODE REVIEW

File: doc/Development-Guide-Angular.md [hunk 1]
```diff
@@ -117,28 +117,45 @@ In addition to those fundamental modules, there are some share modules:
...
--   **appBaseUrl**: Used to configure base address of the client
+-   **appBaseUrl**: Used to configure the base address of the client
     application. Default value: http://localhost:4200
 
-**appBaseUrl** is configured since we use it to define format of our
-URL. If we want to use tenancy name as subdomain for a multi-tenant
+The purpose of **appBaseUrl** is to enable the Angular client to decompose
+the browser address into tenant-name + base-url. 
+
+#### Subdomain Multi-Tenancy:
```
Comment (acjh) [inline doc/Development-Guide-Angular.md]: - Remove colon `:` from heading.
- Add empty line after heading.

File: doc/Development-Guide-Angular.md [hunk 2]
```diff
@@ -117,28 +117,45 @@ In addition to those fundamental modules, there are some share modules:
...
 http://**{TENANCY\_NAME}**.mydomain.com
 
 {TENANCY\_NAME} is the place holder here for tenant names. Tenancy name
-can also be configured for **remoteServiceBaseUrl** as similar. To make
-tenancy name subdomains properly work, we should also make two
-configurations beside the application:
+must be similarly configured for **remoteServiceBaseUrl**. 
+
+It's important to understand that subdomain multi-tenancy is a function
+of the `DomainTenantResolveContributor` ***server-side*** module.
```
Comment (acjh) [inline doc/Development-Guide-Angular.md]: No need to bold and emphasise "server-side" as it already repeats that the class (which only exists server-side) is server-side.

And it isn't a module, so "server-side \`DomainTenantResolveContributor`" is more appropriate.

File: doc/Development-Guide-Angular.md [hunk 3]
```diff
@@ -117,28 +117,45 @@ In addition to those fundamental modules, there are some share modules:
...
+This means that the server API **must** be accessed via a sub-domain that
+conforms to the format you have defined during your configuration of 
+AspNetBoilerplate (see 
+[these notes](https://github.com/aspnetboilerplate/aspnetboilerplate/blob/dev/doc/WebSite/Multi-Tenancy.md#determining-current-tenant) regarding preinitialization).
```
Comment (acjh) [inline doc/Development-Guide-Angular.md]: style as "... during your configuration of ASP.NET Boilerplate \[Multi-Tenancy](...)."

File: doc/Development-Guide-Angular.md [hunk 4]
```diff
@@ -117,28 +117,45 @@ In addition to those fundamental modules, there are some share modules:
...
+<img src="images/angular2-core-subdomain-tenant-resolution-sequence.png" alt="Angular ASP.NET Core Sub-Domain Tenancy Overview" class="img-thumbnail" width="540" height="379" />
+
+To complete the setup of subdomain tenancy, we should also make two
+infrastructural configurations:
 
 1.  We should configure DNS to redirect all subdomains to a static IP
     address. To declare 'all subdomains', we can use wildcard like
-    **\*.mydomain.com**.
+    **\*.mydomain.com**. Note that if you have a split Angular/API
+    setup, then you will need 2 subdomain formats, such as:
```
Comment (acjh) [inline doc/Development-Guide-Angular.md]: - Remove colon `:` from this line and add period `.` at the end of the following line.

**Comment by Adam-Langley** [comment]
@acjh - good spotting - very odd how that happened. Thanks - done.

**Comment by acjh** [comment]
Did you see my other comments?

File: doc/Development-Guide-Angular.md (hunk 1 above)
Comment (Adam-Langley) [reply doc/Development-Guide-Angular.md]: Thanks - done.

File: doc/Development-Guide-Angular.md (hunk 2 above)
Comment (Adam-Langley) [reply doc/Development-Guide-Angular.md]: Thanks - done.

File: doc/Development-Guide-Angular.md (hunk 4 above)
Comment (Adam-Langley) [reply doc/Development-Guide-Angular.md]: Thanks - done.

File: doc/Development-Guide-Angular.md (hunk 3 above)
Comment (Adam-Langley) [reply doc/Development-Guide-Angular.md]: Thanks - done.

**Comment by Adam-Langley** [comment]
Indeed - just unable to get to them until now. All done - thanks.

**Comment by Adam-Langley** [comment]
Hi @acjh  - just curious if there are any issues blocking this from being pulled?
Thanks.

**Comment by ismcagdas** [comment]
@Adam-Langley sorry, I forgot about this.

**Pull request merged**